EventStore/EventStore/Service/AppointmentSchedulingEventService.cs

```diff
@@ -0,0 +1,175 @@
+                return new CountStepsEventDto(GetEventByAttempt(maxEvent.Attempt), maxEvent.Count);
+            }
+            return null;
+        }
+
+        public double GetSuccessfulAttemptsRatio()
+        {
+            return Math.Round(((double)GetSuccessfulAppointmentsAttempts().Count() / (double)GetAll().GroupBy(appointmentEvent => appointmentEvent.Attempt).ToList().Count()), 2);
+        }
+
+        public int GetMostCanceledStep()
+        {
+            return GetAll().ToList().FindAll(appointmentEvent => appointmentEvent.Action.Equals("cancel") && !GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).Select(p => p.Step).ToList().Count > 0 ? GetAll().FindAll(appointmentEvent => appointmentEvent.Action.Equals("cancel") && !GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).Select(p => p.Step).ToList().GroupBy(i => i).OrderByDescending(grp => grp.Count()).Select(grp => grp.Key).First() : 0;
+        }
+
+        public double GetAverageStepsForSuccessfulAttempt()
+        {
+            return (GetSuccessfulAppointmentsAttempts().Count > 0) ? (((double)GetAll().FindAll(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).Count() / (double)GetSuccessfulAppointmentsAttempts().Count) - 2) : 0;
+        }
+
+        public double GetAverageStepsForUnsuccessfulAttempt()
+        {
+            return (GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Count() > 0) ? (((double)GetAll().FindAll(appointmentEvent => GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Contains(appointmentEvent.Attempt)).Count() / (double)GetCanceledAppointmentsAttempts().Except(GetSuccessfulAppointmentsAttempts()).Count()) - 1) : 0;
+        }
+
+        public CountStepsEventDto GetStatisticsMinStepsForCancelling()
+        {
+            List<long> unsuccessfulAppointmentsAttempts = GetCanceledAppointmentsAttempts();
+            List<long> successfulAppointmentsAttempts = GetSuccessfulAppointmentsAttempts();
+            if (successfulAppointmentsAttempts.Count > 0)
+            {
+                var minEvent = GetAll().Where(appointmentEvent => unsuccessfulAppointmentsAttempts.Except(successfulAppointmentsAttempts).Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 1 }).ToList().Aggregate((eventWithMinCount, currentEvent) => currentEvent.Count <= eventWithMinCount.Count ? currentEvent : eventWithMinCount);
+                return new CountStepsEventDto(GetEventByAttempt(minEvent.Attempt), minEvent.Count);
+            }
+            return null;
```

Comment:
Write in the form of a ternary operator.
Example:
"var maxEvent = ...;
return ..."